sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -1,34 +1,48 @@
-# Azure.Core public serialization samples
+# Azure.Core Public Serialization Guide
 
-## Using explicit cast
+The latest updates to the `Azure.Core` library have simplified serializing and deserializing of all public Azure models. With the addiion of the `ModelSerializer` class, we provide a public interface that exposes the previously internal serialization code so customers can access it without needing to use reflection or write their own translation. The recommended serialization approach is to use the static `ModelSerializer` class. The `ModelSerializer` class also has `ModelSerializerOptions` which allows the user to set the serialization `ModelSerializerFormat` type (`Json` or `Wire`) and provide custom Serializer types for specific models.
```

Comment:
In the intro, I'd love to see something like:

Problem framing:
- .NET serialization typically uses JsonSerializer in System.Text.Json
- Azure SDK models can't use that because ___
- That means, for Azure SDK models we need to do _____
- IModelSerializable solves this problem

Overview of serialization types:
- Azure SDK models implement IModelSerializer (because we said that above)
- This can be hard to use as such because ___
- To make it easier to serialize Azure SDK models, we provide ModelSerializer, because it _______ (knows how to use these interfaces?  lets us provide a set of options?  gives us good performance?)

After that, key concepts can follow like:
- What if my models use XML?  IXmlModelSerializable
- How can this help me use protocol methods?  DynamicData
- Well, what if I have a model that's super customized and needs Newtonsoft.JSON?  We have "bring your own model"

In general, each of our types should be addressing a business need - we can help customers see how they can use them by telling them why each thing exists.